packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/Stepper.tsx

```diff
@@ -0,0 +1,87 @@
+import * as steps from "./steps";
+
+import React from "react";
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+import { useSelector } from "react-redux";
+import { SliceMachineStoreType } from "@src/redux/type";
+import {
+  selectOpenedStep,
+  selectSetupStatus,
+  selectUserHasAtLeastOneStepMissing,
+  selectUserHasConfiguredAllSteps,
+} from "@src/modules/preview";
+
+import { isLoading } from "@src/modules/loading";
+import { LoadingKeysEnum } from "@src/modules/loading/types";
+import { Frameworks } from "@slicemachine/core/build/src/models";
+import { getFramework, selectPreviewUrl } from "@src/modules/environment";
+
+interface Props {
+  framework: Frameworks | undefined;
+}
+
+export default function Stepper({ framework }: Props): React.ReactElement {
+  if (framework !== Frameworks.next && framework !== Frameworks.nuxt) {
+    return (
+      <p>
+        Framework {framework || "undefined"} is not supported yet. Please use
+        Storybook instead.
+      </p>
+    );
+  }
+
+  const currentSteps = steps[framework];
+
+  const { toggleSetupDrawerStep, checkPreviewSetup } = useSliceMachineActions();
+  const {
+    previewUrl,
+    openedStep,
+    setupStatus,
+    userHasAtLeastOneStepMissing,
+    userHasConfiguredAllSteps,
+    isCheckingSetup,
+  } = useSelector((state: SliceMachineStoreType) => ({
+    openedStep: selectOpenedStep(state),
+    isCheckingSetup: isLoading(state, LoadingKeysEnum.CHECK_PREVIEW),
+    setupStatus: selectSetupStatus(state),
+    previewUrl: selectPreviewUrl(state),
+    framework: getFramework(state),
+    userHasAtLeastOneStepMissing: selectUserHasAtLeastOneStepMissing(state),
+    userHasConfiguredAllSteps: selectUserHasConfiguredAllSteps(state),
+  }));
+
+  const stepNumberWithErrors = [
```

Comment:
Although this is temporary, I think we should create something like a selector that returns an array per framework. It could be stored next to steps

Reply:
As for now this information is only used here I don't think a selector is needed. Selector are used for centralizing logic around store so we can create a selector when we will reuse somewhere else this logic. 

What do you think about it :)

Reply:
Selector was not the right term. I had something like this in mind

```javascript
export {
   steps: [/* ... */],
   withErrorSteps(setupStatus) {
       return [/* ... */]
   }
```

And 

```javascript
    const { steps: currentSteps, withErrorSteps } = steps[framework]
    const stepsWithErrors = withErrorSteps(setupStatus)
```

But it's fine